DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -262,17 +269,22 @@ function UninstallPackage
 
     $env:Path = [Environment]::GetEnvironmentVariable('Path','Machine')
 
-    #Todo: Refactor
-    if (-not ($pParams))
-    {
-        Write-Verbose -Message 'Uninstalling Package Standard'
-        $packageUninstallOuput = choco uninstall $pName -y
+    [string]$chocoParams = "-y"
+    if ($pParams) {
+        $chocoParams += " --params=`"$pParams`""
     }
-    elseif ($pParams)
-    {
-        Write-Verbose -Message "Uninstalling Package with params $pParams"
-        $packageUninstallOuput = choco uninstall $pName --params="$pParams" -y
+    if ($pVersion) {
+        $chocoParams += " --version=`"$pVersion`""
     }
+    # Check if Chocolatey version is Greater than 0.10.4, and add --no-progress 
+    $chocoVer = [system.version](Get-ChocoInstalledPackage | Where-Object { $_.name -eq 'chocolatey' }).Version

```

Comment:
Just run `choco -v` and inspect the output. Do keep in mind you can't do a System.Version on prereleases, so this would fail.

Reply:
Can you explain why [System.Version] would not work on prereleases?  I thought this was just a .net type fore version numbers that allows you to compare software versions with the assumption that there is a Major.Minor.Build.Revision numbering schema. 

If you set `$ChocoVer = '0.10.4'` it will auto cast it to type Version, I just wanted to be explicit to make sure that it did not end up as a string, as powershell would just be comparing string length if that happened.

Reply:
The version would be `0.10.9-beta1` - cast that to a System.Version and check out the results. However, that Semantic Version is correct, it just doesn't cast to System.Version. You can split the number by the `-` and use the left side (index 0) for the cast into System.Version. That would work.